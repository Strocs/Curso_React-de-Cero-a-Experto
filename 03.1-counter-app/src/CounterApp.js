import PropTypes from 'prop-types'
import { useState } from 'react'

const CounterApp = ({ value }) => {
	const [counter, setCounter] = useState(value)

	//handleAdd
	//setCounter( (c) => c + 1 )
	const   handleAdd = () => setCounter(counter + 1),
		    handleSubtract = () => setCounter(counter - 1),
		    handleReset = () => setCounter(value)

	return (
		<>
			<h1>CounterApp:</h1>
			<h2>{counter}</h2>

			<button onClick={ handleAdd }>+1</button>
			<button onClick={ handleReset }>Reset</button>
			<button onClick={ handleSubtract }>-1</button>
		</>
	)
}

CounterApp.propTypes = {
	value: PropTypes.number.isRequired,
}
CounterApp.defaultProps = {
	value: 0,
}

export default CounterApp
