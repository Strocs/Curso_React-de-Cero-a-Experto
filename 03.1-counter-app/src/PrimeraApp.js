//FC
// import { Fragment } from "react"
import PropTypes from 'prop-types'

// const PrimeraApp = ({ saludo, subtitulo = 'Soy un subtitulo' }) => {
const PrimeraApp = ({ saludo, subtitulo }) => {
	// if(!saludo) {
	//     throw new Error('El saludo es necesario')
	// }

	//Lo mismo que poner Fragments
	return (
		<>
			<h1>{saludo}</h1>
			{/* <pre>{ JSON.stringify(saludo, null, 3)}</pre> */}
			<p>{subtitulo}</p>
		</>
	)
	//Fragment evita que aparezca tenga que encerrar en Div
	// return (
	//     < Fragment>
	//         <h1>Hola Mundo</h1>
	//         <p>Mi primera aplicación</p>
	//     </ Fragment>
	// )
	//para poner más de una etiqueta hay que encerrarla en div
	// return (
	//     <div>
	//         <h1>Hola Mundo</h1>
	//         <p>Mi primera aplicación</p>
	//     </div>
	// )
}

//Forma de obligar a otors que utilicen mis componentes a
//entender qué tipo de valores son necesarios para que
//el componente funcione correctamente
PrimeraApp.propTypes = {
	saludo: PropTypes.string.isRequired,
}

PrimeraApp.defaultProps = {
	subtitulo: 'Mi primera App de React',
}

export default PrimeraApp
