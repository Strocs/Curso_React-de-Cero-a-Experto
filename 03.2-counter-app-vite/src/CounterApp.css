#root {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
}

h1 {
  letter-spacing: .4rem;
  background: linear-gradient(90deg, cyan, blueviolet);
  background-clip: text;
  -webkit-background-clip: text;
  color: transparent;
}

h2 {
  color:lightskyblue;
  font-size: 3rem;
}
button {
  background: aliceblue;
  color: black;
  margin: 1rem;
}
