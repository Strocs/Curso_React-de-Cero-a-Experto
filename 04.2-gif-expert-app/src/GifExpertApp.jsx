/* eslint-disable no-undef */
import { useState } from 'react'
import { AddCategory, GifGrid } from './components'
export const GifExpertApp = () => {
  const [categories, setCategories] = useState([])

  const onAddCategory = newCategory => {
    const categoryIsIncluded = categories.find(
      category => category.toLowerCase() === newCategory.toLowerCase()
    )
    if (categoryIsIncluded) return alert('La categor√≠a ya existe en la lista')
    setCategories([newCategory, ...categories])
  }

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory onAddCategory={onAddCategory} />
      {categories.map(category => (
        <GifGrid key={category} category={['one punch', {}]} />
      ))}
    </>
  )
}
