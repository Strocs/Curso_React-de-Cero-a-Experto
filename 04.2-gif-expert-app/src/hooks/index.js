export * from './useCamelCase'
export * from './useFetchGifs'
