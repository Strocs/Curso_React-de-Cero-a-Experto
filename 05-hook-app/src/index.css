body {
  padding: 20px;
}

button {
  margin-right: 10px;
}

a {
  margin-right: 5px;
}

h2 {
  font-size: 1.5rem;
}