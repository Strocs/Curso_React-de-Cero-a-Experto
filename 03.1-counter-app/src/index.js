// import React from 'react'
// import PrimeraApp from './PrimeraApp'
import ReactDOM from 'react-dom'
import CounterApp from './CounterApp'

import './index.css'

// const saludo = <h1>Hola Mundo</h1>

const divRoot = document.querySelector('#root')

//Qué quiero poner, dónde lo quiero poner
// ReactDOM.render(<PrimeraApp saludo='Hola Mundo' />, divRoot)

// ReactDOM.render(<PrimeraApp saludo='Hwola soy Ignacio' />, divRoot)
ReactDOM.render(<CounterApp />, divRoot)
