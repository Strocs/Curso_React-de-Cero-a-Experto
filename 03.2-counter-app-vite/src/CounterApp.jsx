import React, { useState } from 'react'
import PropTypes from 'prop-types'
import './CounterApp.css'

const CounterApp = ({ value = 0 }) => {
  const [count, setCount] = useState(value)

  return (
    <>
      <h1>Counter App</h1>
      <h2>{count}</h2>
      <div>
        <button onClick={() => setCount(count + 1)} type='button'>
          +1
        </button>
        <button onClick={() => setCount(count - 1)} type='button'>
          -1
        </button>
        <button
          aria-label='btn-reset'
          onClick={() => setCount(value)}
          type='button'
        >
          Reset
        </button>
      </div>
    </>
  )
}

export default CounterApp

CounterApp.propTypes = {
  value: PropTypes.number
}
