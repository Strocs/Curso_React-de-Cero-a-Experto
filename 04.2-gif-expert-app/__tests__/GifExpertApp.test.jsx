import { render, screen } from '@testing-library/react'
import { describe, expect, test } from 'vitest'
import { GifExpertApp } from '../src/GifExpertApp'

describe('tests in <GifExpertApp />', () => {
  test('should match with snapshot of initial state', () => {
    const { container } = render(<GifExpertApp />)
    expect(container).toMatchSnapshot()
  })

  test('should return an alert if category is included', () => {
    render(<GifExpertApp />)
    screen.debug()
  })
})
