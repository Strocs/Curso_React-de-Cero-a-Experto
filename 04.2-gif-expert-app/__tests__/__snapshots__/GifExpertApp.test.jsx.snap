// Vitest Snapshot v1

exports[`tests in <GifExpertApp /> > should match with snapshot of initial state 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gif"
      type="text"
      value=""
    />
  </form>
</div>
`;
